galaxy/api/views/repository.py

```diff
@@ -51,6 +52,16 @@ def get_repo(provider_namespace, user, repo_name):
     return repo
 
 
+def check_name(name):
+    if not name:
+        raise ValidationError(detail={'name': 'Value required'})
+
+    if re.search('[^A-Za-z0-9-_]', name):
```

Comment:
IMHO `re.match('^[\w-]+$')` is slightly better for readability, but this code is totally fine.